### PR TITLE
Add a pipeline to request doc reviewers

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Add a pipeline that runs on all repos with the webhook installed.
This pipeline requests the @tektoncd/docs.reviewers team as reviewer
if files changed in the PR match "(^docs\/.*|^(?!vendor\/)(?!thirdparty\/).*\.(md|rst)$)"

Note that even if the team is only added to the list of reviewers if
it is a collaborator of the repo, a notification email is sent to
the team regardless.

Update the DNS docs in the main README

Signed-off-by: Andrea Frittoli <andrea.frittoli@uk.ibm.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

/kind feature

### CODE REVIEW

**Comment by afrittoli** [comment]
This is still WIP
/hold

File: tekton/ci-workspace/jobs/tekton-docs-reviews.yaml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  - name: set-reviewers
+    image: python:3.9-alpine3.13
+    script: |
+      #!/usr/bin/env python
+      import json
+      import http.client
+
+      # Load the GitHub Token
+      github-token = ""
+      with open("$(workspaces.github.path/$(params.github-token-key", "r") as gh:
```
Comment (sbwsg) [inline tekton/ci-workspace/jobs/tekton-docs-reviews.yaml:36]: `$(workspaces.github.path` -> I think you are missing a `)` here?
Similarly with `$(params.github-token-key`.

Comment (afrittoli) [reply tekton/ci-workspace/jobs/tekton-docs-reviews.yaml:36]: Nice catch, thank you!

File: tekton/ci-workspace/jobs/tekton-docs-reviews.yaml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        github-token = gh.read()
+
+      # Implement the request reviewes GitHub API
```
[the remark this answers is not in the document]
Comment (afrittoli) [reply tekton/ci-workspace/jobs/tekton-docs-reviews.yaml:39]: Thank you!

**Review by sbwsg — APPROVED** [review]

File: tekton/ci-workspace/jobs/tekton-docs-reviews.yaml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+spec:
+  params:
+    - name: package
+      description: "{owner}/{repo}"
+    - name: pullRequestNumber
+      description: The pull request number
+    - name: github-token-key
+      description: Name of the GitHub token key / file
+    - name: reviewers
+      description: Command separated list of github user slugs
```
Comment (mpetason) [inline tekton/ci-workspace/jobs/tekton-docs-reviews.yaml:17]:
```suggestion
      description: Comma separated list of github user slugs
```

File: tekton/ci-workspace/jobs/tekton-docs-reviews.yaml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+      default: ""
+    - name: team-reviewers
+      description: Command separated list of github team slugs
```
Comment (mpetason) [inline tekton/ci-workspace/jobs/tekton-docs-reviews.yaml:20]:
```suggestion
      description: Comma separated list of github team slugs
```

**Review by mpetason — COMMENTED** [review]
I think these are comma separated lists - but I could be wrong. Updated just in case.

**Comment by afrittoli** [comment]
Test comment

**Comment by afrittoli** [comment]
Test comment

**Comment by afrittoli** [comment]
Test comment

**Comment by afrittoli** [comment]
Test comment

**Comment by afrittoli** [comment]
Test comment

**Comment by afrittoli** [comment]
Using the test from the catalog now: https://hub.tekton.dev/tekton/task/github-request-reviewers

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**